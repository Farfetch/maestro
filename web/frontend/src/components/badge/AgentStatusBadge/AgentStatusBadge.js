import { Badge } from "antd";
import PropTypes from "prop-types";
import React from "react";

import { agentStatus as agentStatusModel } from "../../../lib/api/models";

/**
 *
 * @param {string} agentStatus
 * @returns string
 * https://ant.design/components/badge/#components-badge-demo-status
 */
const getBadgeStatus = (agentStatus) => {
  if (agentStatus === agentStatusModel.CREATING) return "warning";
  if (agentStatus === agentStatusModel.AVAILABLE) return "success";
  if (agentStatus === agentStatusModel.UNAVAILABLE) return "error";
  if (
    [agentStatusModel.PROCESSING_EVENT, agentStatusModel.RUNNING_TEST].includes(
      agentStatus
    )
  )
    return "processing";

  return "default";
};

const AgentStatusBadge = ({ agentStatus, text = "" }) => (
  <>
    <Badge status={getBadgeStatus(agentStatus)} text={text} />
  </>
);

AgentStatusBadge.propTypes = {
  agentStatus: PropTypes.oneOf(Object.values(agentStatusModel)).isRequired,
  text: PropTypes.string.isRequired
};

export default AgentStatusBadge;
