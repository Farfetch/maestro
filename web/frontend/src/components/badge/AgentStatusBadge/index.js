import AgentStatusBadge from "./AgentStatusBadge";

export default AgentStatusBadge;
