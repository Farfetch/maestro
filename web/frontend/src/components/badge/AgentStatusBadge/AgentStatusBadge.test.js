import { render } from "@testing-library/react";
import React from "react";

import AgentStatusBadge from "./AgentStatusBadge";

describe("components/AgentStatusBadge", () => {
  [
    "CREATING",
    "AVAILABLE",
    "PROCESSING_EVENT",
    "RUNNING_TEST",
    "UNAVAILABLE"
  ].map((agentStatus) =>
    test(`should render ${agentStatus} badge`, async () => {
      const rendered = render(
        <AgentStatusBadge agentStatus={agentStatus} text={agentStatus} />
      );

      const badgeComponent = rendered.container;
      expect(badgeComponent.outerHTML).toMatchSnapshot();
    })
  );
});
