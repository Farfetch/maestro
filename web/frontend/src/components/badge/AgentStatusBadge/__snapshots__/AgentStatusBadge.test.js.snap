// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AgentStatusBadge should render AVAILABLE badge 1`] = `"<div><span class=\\"ant-badge ant-badge-status ant-badge-not-a-wrapper\\"><span class=\\"ant-badge-status-dot ant-badge-status-success\\"></span><span class=\\"ant-badge-status-text\\">AVAILABLE</span></span></div>"`;

exports[`components/AgentStatusBadge should render CREATING badge 1`] = `"<div><span class=\\"ant-badge ant-badge-status ant-badge-not-a-wrapper\\"><span class=\\"ant-badge-status-dot ant-badge-status-warning\\"></span><span class=\\"ant-badge-status-text\\">CREATING</span></span></div>"`;

exports[`components/AgentStatusBadge should render PROCESSING_EVENT badge 1`] = `"<div><span class=\\"ant-badge ant-badge-status ant-badge-not-a-wrapper\\"><span class=\\"ant-badge-status-dot ant-badge-status-processing\\"></span><span class=\\"ant-badge-status-text\\">PROCESSING_EVENT</span></span></div>"`;

exports[`components/AgentStatusBadge should render RUNNING_TEST badge 1`] = `"<div><span class=\\"ant-badge ant-badge-status ant-badge-not-a-wrapper\\"><span class=\\"ant-badge-status-dot ant-badge-status-processing\\"></span><span class=\\"ant-badge-status-text\\">RUNNING_TEST</span></span></div>"`;

exports[`components/AgentStatusBadge should render UNAVAILABLE badge 1`] = `"<div><span class=\\"ant-badge ant-badge-status ant-badge-not-a-wrapper\\"><span class=\\"ant-badge-status-dot ant-badge-status-error\\"></span><span class=\\"ant-badge-status-text\\">UNAVAILABLE</span></span></div>"`;
